src/PointwiseFunctions/SpecialRelativity/LorentzBoostMatrix.cpp

```diff
@@ -0,0 +1,109 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/SpecialRelativity/LorentzBoostMatrix.hpp"
+
+#include <cmath>
+#include <cstddef>
+#include <limits>
+
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace sr {
+template <size_t SpatialDim, typename Frame>
+tnsr::Ab<double, SpatialDim, Frame> lorentz_boost_matrix(
+    const tnsr::I<double, SpatialDim, Frame>& velocity) noexcept {
+  auto boost_matrix = make_with_value<tnsr::Ab<double, SpatialDim, Frame>>(
+      get<0>(velocity), std::numeric_limits<double>::signaling_NaN());
+  lorentz_boost_matrix<SpatialDim, Frame>(&boost_matrix, velocity);
+  return boost_matrix;
+}
+
+template <size_t SpatialDim, typename Frame>
+void lorentz_boost_matrix(
+    gsl::not_null<tnsr::Ab<double, SpatialDim, Frame>*> boost_matrix,
+    const tnsr::I<double, SpatialDim, Frame>& velocity) noexcept {
+  const double velocity_squared{get(dot_product(velocity, velocity))};
+  const double lorentz_factor{1.0 / sqrt(1.0 - velocity_squared)};
+
+  // For the spatial-spatial terms of the boost matrix, we need to compute
+  // a prefactor, which is essentially kinetic energy per mass per velocity
+  // squared. Specifically, the prefactor is
+  //
+  // kinetic_energy_per_v_squared = (lorentz_factor-1.0)/velocity^2
+  //
+  // We would like to avoid large numerical errors (or even FPEs) when
+  // the velocity is zero or close to zero. The idea is to use a truncated
```

Comment:
Would using `kinetic_energy_per_v_squared = 1 / ((1 + sqrt(1-v^2)) sqrt(1-v^2))` be better behaved?